### DISCUSSION TITLE
Error feedback during Loras merge.

### DISCUSSION THREAD

**Discussion by newseanstation** [hako-mikan/sd-webui-supermerger#374] [General]
When merging the SDXL version of Loras, the following information was provided as feedback.：

merging D:\05newSD\stable-diffusion-webui\models\Lora\SDXL\chubbyladyXL2.0.safetensors: [0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5]
dim: [32], alpha: [32.0]
merging...
Traceback (most recent call last):
  File "D:\05newSD\stable-diffusion-webui\extensions\sd-webui-supermerger\scripts\mergers\pluslora.py", line 426, in lmerge
    sd = merge_lora_models(ln, lr, settings, False, calc_precision)
  File "D:\05newSD\stable-diffusion-webui\extensions\sd-webui-supermerger\scripts\mergers\pluslora.py", line 483, in merge_lora_models
    ratio = ratios[blockfromkey(key, keylist, isv2)]
IndexError: list index out of range

how to fix it? 
thanks

**Comment by Mako-95** [comment]
<img width="999" alt="err1" src="https://github.com/hako-mikan/sd-webui-supermerger/assets/136629443/7560c39e-b8cd-438c-a712-72857f2915e4">
While not a proper solution, there is a way to circumvent the issue.

If You go to the file from the error message and find the 'ratio = ratios[blockfromkey(key, keylist, isv2)]' line, You can force it to work by changing it to 'ratio = ratios[min(blockfromkey(key, keylist, isv2), len(ratios)-1)]'.
This ensures that the index of the for loop does not go over the maximum array index.

I did that, got the same error on a different line (546) with the same problem.
<img width="1024" alt="err2" src="https://github.com/hako-mikan/sd-webui-supermerger/assets/136629443/b1701ba0-8db5-4059-bcfb-d69ec2689461">
Did the same "fix", keeping the new arguments of 'blockfromkey' (so the LBLCOKS26 rather then keylist, the rest was the same as above), and now the merger works even with SDXL LoRAs.

Again, this is just a band-aid fix, not a solution (since I don't know the actual source of the problem), but the resulting LoRA does what I wanted it to, so You could try doing the same.
